### ISSUE TITLE
Create examples

### ISSUE THREAD

**Issue by xRadne** [strusoft/femdesign-api#34]
Create examples for user to get started (and explore possibilites) with the API.

To start with, **all** the examples should be made for Grasshopper. Preferably all examples should have an equivalent in C#, Dynamo as well, but for now only **the most basic** examples should be created. 

### Instructions on getting started
- [x] (Maybe) Update `README.md` installation section, if something is unclear.

### Basic examples to implement (Scripts)
- [x] **Creating** a simply supported **beam** with supports, loads, loadcases - prio 1
- [x] **Analyse** a **simple** model - prio 1
- [ ]  **Analyse** a model with advanced settings for loadcombinations (Freq. NLE. etc.) - prio 1
- [ ] **Deconstruct** a model, edit and reconstruct - prio 1
- [ ] **Creating custom section** in FEM-Design and load using `SectionDatabase.FromStruxml`
- [ ] **Autodesign** an RC-beam/wall/structure.
- [ ] **Creating** a **wall** with supports, loads, loadcases
- [ ] _To be added_

### Practical examples to implement (Scripts)
- [x] **Creating** a parametric **bridge** model similar to this - **prio 1** See attached specification below
![image](https://user-images.githubusercontent.com/89396751/149291052-89f362d3-5ec8-4694-9a8c-337a035f4da7.png)
http://blog.rhino3d.com/2021/10/parametric-bridge-design-online.html
https://sofistikforyou.com/challenges-of-bridge-design-and-detailing-in-autodesk-revit/
[Exempel Bro Specifikation.docx](https://github.com/strusoft/femdesign-api/files/7924633/Exempel.Bro.Specifikation.docx)
- [ ] Create an example generating a simple plate (balcony?) and loads, running analysis and deciding reinforcement **(iterative analysis)**
- [ ] **Mapping** between Revit/FEM Design (already started)
- [ ] Example with iterative analysis in Grasshopper, using maybe Anemone or similar (for users who don't know C#)

### Wishes from customers
- [ ] Reading results
- [ ] Adding elements to an existing model
- [ ] Railway noise barriers
- [ ] Basic piling situation

**Comment by JRiad** [comment]
I think we need to have
1. Basic examples for understanding and learning how to use the API
2. Practical examples on what you actually want to do with the API, "real world problems"

Some (relatively simple) real world problems should be prioritized as they explain why you should even use the API

**Comment by JRiad** [comment]
@xRadne and @samnyb , I guess we should update examples to FD 21 version. Because they were made for FD 20 right?

**Comment by xRadne** [comment]
I agree! However it is not required since the API is backward compatible, meaning all `grasshopper` scripts made in v20 can be used with API v21.

The same goes for `Dynamo`. 

But not `C#`, but it will give clear hints on what needs to be fixed,

**Comment by xRadne** [comment]
> I agree! However it is not required since the API is backward compatible, meaning all `grasshopper` scripts made in v20 can be used with API v21.
> 
> The same goes for `Dynamo`.
> 
> But not `C#`, but it will give clear hints on what needs to be fixed,

Include this in documentation #35. Ping @andosca

**Comment by xRadne** [comment]
@samnyb I think the About component #189 should be added to all scripts as well.

![image](https://user-images.githubusercontent.com/32938476/153141833-44828933-c20e-4bb4-b0f1-e99efe95d025.png)

**Comment by samnyb** [comment]
> @xRadne and @samnyb , I guess we should update examples to FD 21 version. Because they were made for FD 20 right?

> @samnyb I think the About component #189 should be added to all scripts as well.

I'll put both these things on my to-do-list!

**Comment by andosca** [comment]
> > I agree! However it is not required since the API is backward compatible, meaning all `grasshopper` scripts made in v20 can be used with API v21.
> > The same goes for `Dynamo`.
> > But not `C#`, but it will give clear hints on what needs to be fixed,
> 
> Include this in documentation #35. Ping @andosca

There is not any clear distinction between FD21 and FD20. For now it is backwards compatible as no xml-schema changes were introduced.

**Comment by JRiad** [comment]
Is it really backwards compatible all the way? I thought some components may have a changed number of inputs or have some changes, making it necessary to replace them.  I think it would be nice at least to check that they work in FD 21

**Comment by xRadne** [comment]
> Is it really backwards compatible all the way? I thought some components may have a changed number of inputs or have some changes, making it necessary to replace them. I think it would be nice at least to check that they work in FD 21

If the grasshopper plugin is not backwards compatible. It's a bug. The components we have changed should be functioning but marked as `OLD`.

**Comment by samnyb** [comment]
> > @xRadne and @samnyb , I guess we should update examples to FD 21 version. Because they were made for FD 20 right?
> 
> > @samnyb I think the About component #189 should be added to all scripts as well.
> 
> I'll put both these things on my to-do-list!

- [ ] Checked all examples for `OLD` components
- [ ] Added About component to all examples

**Comment by samnyb** [comment]
Possible improvement of bridge example: using the `Perp Frames` component instead of orienting planes along axis. Cuts 7-8 components per flange. I haven't tested if it's a complete match in functionality though, so I'll have to get back to it at a later point.

**Comment by samnyb** [comment]
**22/03/16: Add Python filepath component to all existing examples**

- [ ] Example 1
- [ ] Example 2
- [ ] Example 3
- [ ] Example 4
- [ ] Example 5
- [ ] Example 6 (in progress)
- [ ] Potentially the Bridge example

**Comment by samnyb** [comment]
2022-03-24: Load combination factors apparently in wrong order in example 2.
- [ ] Check example 2
- [ ] Check other examples for similar error

**Comment by xRadne** [comment]
We should create these examples in `C#` as well. These examples should be equivalent so that a user familiar with one framework could easily compare c# to grasshopper (and eventually python and/or dynamo)

**Comment by samnyb** [comment]
We have initiated the process of migrating examples form FemDesign.Samples to a future directory called FemDesign.Examples. Progress on this is mapped in #203.

**Comment by samnyb** [comment]
@xRadne when you have time, I would like to merge branch 34 to master to set Example 7 in the right folder, and also rebase (?) branch 34 so the new structure is applied there as well, and all duplicate files removed. I still feel a little unsure around merging, so I would like your support, as well as a green light of the whole process, of course.

**Comment by xRadne** [comment]
> @xRadne when you have time, I would like to merge branch 34 to master to set Example 7 in the right folder, and also rebase (?) branch 34 so the new structure is applied there as well, and all duplicate files removed. I still feel a little unsure around merging, so I would like your support, as well as a green light of the whole process, of course.

Example 7 added to master branch
